packages/runtime/container-runtime/src/test/garbageCollection.spec.ts

```diff
@@ -257,323 +231,412 @@ describe("Garbage Collection Tests", () => {
             defaultGCData.gcNodes[nodes[3]] = [nodes[0]];
         });
 
-        it("doesn't generate events for referenced nodes", async () => {
-            const garbageCollector = createGarbageCollector();
-
-            // Run garbage collection on the default GC data where everything is referenced.
-            await garbageCollector.collectGarbage({ runGC: true });
-
-            // Validate that no inactive events are logged yet.
-            await updateAllNodesAndRunGC(garbageCollector);
-            validateNoInactiveEvents();
-
-            // Expire the unreferenced timer (if any).
-            clock.tick(inactiveTimeoutMs + 1);
+        const tests = (
+            timeout: number,
+            revivedEventName: string,
+            changedEventName: string,
+            loadedEventName: string,
+            snapshotCacheExpiryMs?: number,
+            deleteEventName?: string,
+        ) => {
+            // Validates that no unexpected event has been fired.
+            function validateNoUnexpectedEvents() {
+                assert(
```

Comment:
nit: use `mockLogger.assertMatchAny`

Reply:
And throughout.  It prints out the expected/actual when they don't match which is pretty useful.

Reply:
Can't do it here because its validating doesn't match any. Updated this in other places.